### PR TITLE
Keycloak: Fix build script and maven shading

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR fixes the build.sh to exclude some modules and also activate the shade plugin my slighly in build modification of the pom.xml.

### CODE REVIEW

File: projects/keycloak/build.sh [hunk 1]
```diff
@@ -18,24 +18,71 @@
...
 # Build Keycloak
+
+## Maven build arguments
 MAVEN_ARGS="-Djavac.src.version=17 -Djavac.target.version=17 "
 MAVEN_ARGS=$MAVEN_ARGS"-DskipTests -Dgpg.skip -Dmaven.source.skip "
-MAVEN_ARGS=$MAVEN_ARGS"-DskipExamples -DskipTestsuite"
-$MVN clean package $MAVEN_ARGS
+MAVEN_ARGS=$MAVEN_ARGS"-DskipExamples -DskipTestsuite -DskipQuarkus"
+
+## Exclude unfuzzed modules
```
Comment (AdamKorcz) [inline projects/keycloak/build.sh:32]: Can you add a comment why the build script does this?

File: projects/keycloak/build.sh [hunk 2]
```diff
@@ -18,24 +18,71 @@
...
+EXCLUDE_QUARKUS=$EXCLUDE_QUARKUS"!quarkus/server,!quarkus/dist,!quarkus/tests,!quarkus/tests/junit5"
+
+EXCLUDE_REST="!rest,!rest/admin-ui-ext"
+
+EXCLUDE_SERVICE="!services"
+
+EXCLUDE_MODULE=$EXCLUDE_DOCS,$EXCLUDE_DEPENDENCY,$EXCLUDE_FEDERATION,$EXCLUDE_INTEGRATION,$EXCLUDE_JS
+EXCLUDE_MODULE=$EXCLUDE_MODULE,$EXCLUDE_MISC,$EXCLUDE_MODEL,$EXCLUDE_QUARKUS,$EXCLUDE_REST
+
+## Activate shade plugin
```
Comment (AdamKorcz) [inline projects/keycloak/build.sh:61]: Can you add a comment why the build script does this?

**Review by AdamKorcz — CHANGES_REQUESTED** [review]

File: projects/keycloak/build.sh (hunk 1 above)
Comment (arthurscchan) [reply projects/keycloak/build.sh:32]: done.

**Review by AdamKorcz — APPROVED** [review]
LGTM

File: projects/keycloak/build.sh [hunk 3]
```diff
@@ -18,24 +18,76 @@
...
+for JARFILE in $(find ./ -name "*.jar")
 do
   if [[ "$JARFILE" == *"core/"* ]] || [[ "$JARFILE" == *"saml-core/"* ]] || \
   [[ "$JARFILE" == *"saml-core-api/"* ]] || [[ "$JARFILE" == *"common/"* ]] || \
-  [[ "$JARFILE" == *"jboss-log"* ]] || [[ "$JARFILE" == *"jackson"* ]]
+  [[ "$JARFILE" == *"crypto/"* ]]
   then
-    cp $JARFILE $OUT/
-    RUNTIME_CLASSPATH=$RUNTIME_CLASSPATH\$this_dir/$(basename $JARFILE):
+    if [[ "$JARFILE" != *"original"* ]]
```
[the remark this answers is not in the document]
Comment (arthurscchan) [reply projects/keycloak/build.sh:88]: Done

**Pull request merged**